### PR TITLE
Upgrade the version of the dependency controller-runtime from `v0.1.10` to `v0.1.12`

### LINKED ISSUE #1606: upgrade controller-runtime to support LabelSelector in fakeClient List func
## Feature Request

**Is your feature request related to a problem? Please describe.**

Currently the sdk is using `controller-runtime` `v0.1.10`, which does not support LabelSelector filter in fakeClient List func, only support Namespace filter

https://github.com/operator-framework/operator-sdk/blob/3cf985d4cbc5cebf14430079d3c72176f9866ed5/Gopkg.toml#L33-L35

https://github.com/kubernetes-sigs/controller-runtime/blob/12d98582e72927b6cd0123e2b4e819f9341ce62c/pkg/client/fake/client.go#L91-L114

```go
func (c *fakeClient) List(ctx context.Context, opts *client.ListOptions, list runtime.Object) error {
	gvk, err := getGVKFromList(list, c.scheme)
	if err != nil {
		// The old fake client required GVK info in Raw.TypeMeta, so check there
		// before giving up
		if opts.Raw == nil || opts.Raw.TypeMeta.APIVersion == "" || opts.Raw.TypeMeta.Kind == "" {
			return err
		}
		gvk = opts.Raw.TypeMeta.GroupVersionKind()
	}

	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
	**o, err := c.tracker.List(gvr, gvk, opts.Namespace)**
	if err != nil {
		return err
	}
	j, err := json.Marshal(o)
	if err != nil {
		return err
	}
	decoder := scheme.Codecs.UniversalDecoder()
	_, _, err = decoder.Decode(j, nil, list)
	return err
}
```

**Describe the solution you'd like**

here I request to upgrade to at least `v0.1.11` to support the LabelSelector in fakeClient List

https://github.com/kubernetes-sigs/controller-runtime/blob/477bf4f046c31c351b46fa00262bc814ac0bbca1/pkg/client/fake/client.go#L93-L123

```go
func (c *fakeClient) List(ctx context.Context, opts *client.ListOptions, list runtime.Object) error {
	gvk, err := getGVKFromList(list, c.scheme)
	if err != nil {
		// The old fake client required GVK info in Raw.TypeMeta, so check there
		// before giving up
		if opts.Raw == nil || opts.Raw.TypeMeta.APIVersion == "" || opts.Raw.TypeMeta.Kind == "" {
			return err
		}
		gvk = opts.Raw.TypeMeta.GroupVersionKind()
	}

	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
	o, err := c.tracker.List(gvr, gvk, opts.Namespace)
	if err != nil {
		return err
	}
	j, err := json.Marshal(o)
	if err != nil {
		return err
	}
	decoder := scheme.Codecs.UniversalDecoder()
	_, _, err = decoder.Decode(j, nil, list)
	if err != nil {
		return err
	}

	**if opts.LabelSelector != nil {
		return filterListItems(list, opts.LabelSelector)
	}**
	return nil
}
```

I am willing to help submitting a PR for this upgrade if required, thanks

### PR DESCRIPTION
**Pull request by andy2046**
**Description of the change:**

***Changed***

-  Upgrade the version of the dependency [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) from `v0.1.10` to `v0.1.12`

**Motivation for the change:**
Address the issue in #1606 

Closes #1606

### CODE REVIEW

**Comment by andy2046** [comment]
@LiliC CI is failing in test/e2e/ansible-molecule Scenario: 'test-local' Action: 'prepare', it's my first time running the CI, not sure what causes the error, need your guidance on this

**Comment by LiliC** [comment]
Thanks for the PR, I restarted the failed job, it could be a timeout issue. Also started the other tests, will have a look a bit later.

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 ### Changed
 
+- Upgrade the version of the dependency [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) from `v0.1.10` to `v0.1.12` [#1612](https://github.com/operator-framework/operator-sdk/pull/1612)
```
Comment (LiliC) [inline CHANGELOG.md]:
```suggestion
- Upgrade the version of the dependency [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) from `v0.1.10` to `v0.1.12`. ([#1612](https://github.com/operator-framework/operator-sdk/pull/1612))
```

**Comment by joelanford** [comment]
@andy2046 There are a few more things that need to be covered as well:

- These changes cover the SDK itself, but they will not apply to scaffolded projects. Can you also update the `gopkgtoml.go` and `go_mod.go` files in `internal/pkg/scaffold`, `internal/pkg/scaffold/ansible`, and `internal/pkg/scaffold/helm`
- controller-runtime `0.1.12` also bumped the kubernetes version to `1.13.4`, which we need to do as well, also in all of those files.

The `go.mod` templates have a `// Pinned to kubernetes-1.13.1` section to update and the `Gopkg.toml` templates have overrides for `version` tags or `revision` SHAs that correspond to the `kubernetes-1.13.*` tag.

If you need help figuring out how to change those, let us know and we can help out.

**Comment by andy2046** [comment]
@joelanford thanks for the guidance, I have upgraded Kubernetes to v1.13.4, pls let me know if I missed out any other file.

**Comment by andy2046** [comment]
@LiliC I have updated the docs,  thanks.

**Comment by LiliC** [comment]
@andy2046 seems like some tests actually [failed](https://travis-ci.org/operator-framework/operator-sdk/builds/552109978?utm_source=github_status&utm_medium=notification), doesn't look like a flake. Have you tried running the test locally to be able to debug this? Otherwise, we can help with figuring out the problem here, just let us know! Thanks for this!

**Comment by andy2046** [comment]
@LiliC I can see the error came from `test/subcommand/scorecard` but when I ran `make test/subcommand/scorecard` locally, it passed without error, not sure what's happening there, need your hint to debug, thanks

**Comment by LiliC** [comment]
@andy2046 green now. This was odd, anyways, ready for another round of reviews @joelanford

**Pull request merged**